src/com/facebook/buck/jvm/groovy/GroovyLibraryDescription.java

```diff
@@ -99,7 +106,14 @@ public Arg createUnpopulatedConstructorArg() {
                 /* additionalClasspathEntries */ ImmutableSet.<Path>of(),
                 new GroovycToJarStepFactory(
                     groovyBuckConfig.getGroovyCompiler().get(),
-                    args.extraArguments),
+                    args.extraGroovycArguments,
+                    JavacOptionsFactory.create(
+                        defaultJavacOptions,
+                        params,
+                        resolver,
+                        pathResolver,
+                        args
+                    )),
```

Comment:
If you can figure that out, and it's something we can check into the project, we'd take a PR for that :)